tests/unit/charms/magma_orchestrator_interface/v0/test_magma_orchestrator_interface_requirer.py

```diff
@@ -63,6 +63,37 @@ def test_given_orchestrator_information_in_relation_data_when_relation_changed_t
         self.assertEqual(orchestrator_available_event.fluentd_address, fluentd_address)
         self.assertEqual(orchestrator_available_event.fluentd_port, fluentd_port)
 
+    @patch(f"{BASE_CHARM_DIR}._on_orchestrator_available")
+    def test_given_orchestrator_information_in_relation_data_when_relation_changed_and_schema_validation_fails_then_orchestrator_available_event_is_not_emitted(  # noqa: E501
+        self, patch_on_orchestrator_available
+    ):
+        remote_app = "magma-orc8r-provider"
+        relation_id = self.harness.add_relation(
+            relation_name=self.relation_name, remote_app=remote_app
+        )
+
+        root_ca_certificate = "whatever certificate"
+        wrong_orchestrator_address = "not a url"
+        orchestrator_port = 123
+        bootstrapper_address = "http://bootstrapper.com"
+        bootstrapper_port = 456
+        fluentd_address = "http://fluentd.com"
+        fluentd_port = 789
```

Comment:
The `REQUIRER_JSON_SCHEMA` type for the ports is string. Should the ports be strings instead of integers?

Reply:
They are converted to strings :)
I'm find with keeping it like this, because this is how the library behaves too.
Gets integers and then converts them to strings so it can store them in relation data.

Reply:
My bad. Resolved.